packages/react/src/primitives/Table/TableCell.tsx

```diff
@@ -0,0 +1,26 @@
+import classNames from 'classnames';
+import * as React from 'react';
+
+import { View } from '..';
+import { ComponentClassNames } from '../shared/constants';
+import { TableCellProps } from '../types/table';
+
+export const TableCell: React.FC<TableCellProps> = ({
+  as: asElementTag = 'td',
```

Comment:
I had only separated the components to be more in-line with how one would build a table in native HTML. I felt it would make the components more flexible, but I'm open to changing it so that we could do something like:
```jsx
/* *** */
const TableHeader = ({ cellData, ...rest }) => (
  <thead>
    <tr>
      {cellData.map(cellDatum => <th>{cellDatum}</th>)}
    </tr>
  </thead>
);

<Table>
  <TableHeader cellData=[1, 2, 3]>
</Table>
```